### PR TITLE
[MIG][17.0] scrap_reason_code: Migration to 17.0

### PR DESCRIPTION
**Pull request by john-herholz-dt**

### CODE REVIEW

File: scrap_reason_code/tests/test_scrap_reason_code.py [hunk 1]
```diff
@@ -106,17 +106,17 @@ def test_scrap_reason_code(self):
...
         )
         self.assertEqual(
             scrap.scrap_qty,
             5,
             "Scrap quantity has been modified and is not " "correct anymore.",
         )
-        move = scrap.move_id
-        self.assertEqual(move.reason_code_id.id, self.reason_code.id)
+        moves = scrap.move_ids
+        self.assertEqual(moves.mapped("reason_code_id.id"), [self.reason_code.id])
```
Comment (rousseldenis) [inline scrap_reason_code/tests/test_scrap_reason_code.py:117]: FYI, Odoo supports mapping through all relational records with dot. So, `moves.reason_code_id.id` will work.

**Review by rousseldenis — APPROVED** [review]
LGTM. Code and functional review

File: scrap_reason_code/tests/test_scrap_reason_code.py (hunk 1 above)
Comment (john-herholz-dt) [reply scrap_reason_code/tests/test_scrap_reason_code.py:117]: @rousseldenis Thanks for looking into it.
I know that, but isn't there a technical possibility that moves are more than one record and moves.reason_code_id.id might end up in a singleton error?

edit: ...but maybe not in the context of this test, true.

Comment (rousseldenis) [reply scrap_reason_code/tests/test_scrap_reason_code.py:117]: @john-herholz-dt The only case where you have that is if you put in the chain (without mapped()) another field (Boolean, Selection, Char,...)

**Review by gurneyalex — APPROVED** [review]

**Pull request merged**